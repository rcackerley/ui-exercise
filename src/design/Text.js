// @flow strict @format

import React from "react"
import { theme } from "./theme"

import injectSheet from "react-jss"

import classNames from "classnames"

const styles = {
  text: {
    fontFamily: theme.text.family
  },
  h1: {
    fontSize: "26px",
    fontWeight: 600,
    color: theme.colors.tertiary,
    marginBottom: "15px"
  },
  h2: {
    fontSize: "16px",
    fontWeight: 600,
    color: theme.colors.tertiary
  },
  h3: {
    fontSize: "12px",
    fontWeight: 400,
    color: theme.colors.tertiary
  },
  h4: {
    fontSize: "14px",
    fontWeight: 400,
    color: theme.colors.tertiary
  },
  h5: {
    fontSize: "14px",
    fontWeight: 300,
    color: theme.colors.tertiary
  },
  h6: {
    fontSize: "12px",
    fontWeight: 300,
    letterSpacing: "1px"
  },
  p: {
    fontSize: "12px",
    fontWeight: 400,
    color: theme.colors.black
  },
  link: {
    fontSize: "14px",
    fontWeight: 300,
    color: theme.colors.tertiary
  },
  uppercase: {
    textTransform: "uppercase"
  }
}

type Props = {|
  classes: { [string]: string },
  children: string,
  uppercase: ?boolean,
  variant: string,
  selected: ?boolean,
  deselected: ?boolean,
  customStyles: string
|}

const Text = ({
  variant,
  classes,
  children,
  uppercase,
  customStyles
}: Props) => (
  <span
    className={classNames(
      classes.text,
      classes[variant],
      uppercase && classes.uppercase,
      customStyles
    )}
  >
    {children}
  </span>
)

export default injectSheet(styles)(Text)
