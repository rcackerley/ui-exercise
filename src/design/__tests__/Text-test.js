import React from "react"
import ReactDOM from "react-dom"
import Text from "../Text"

import renderer from "react-test-renderer"

it("render text component correctly", () => {
  const TextComponent = renderer.create(<Text />).toJSON()
  expect(TextComponent).toMatchSnapshot()
})
