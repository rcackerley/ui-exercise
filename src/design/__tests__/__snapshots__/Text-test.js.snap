// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render text component correctly 1`] = `
<span
  className="Text-text-0-1-1"
/>
`;
